lightkube/core/generic_client.py

```diff
@@ -192,6 +193,14 @@ def raise_for_status(resp):
     def build_adapter_request(self, br: BasicRequest):
         return self._client.build_request(br.method, br.url, params=br.params, json=br.data, headers=br.headers)
 
+    def convert_to_resource(self, res: Type[r.Resource], item: dict) -> r.Resource:
+        resource_def = res._api_info.resource
+        if not issubclass(res, dc_d.DataclassDictMixIn):
```

Comment:
Python's type system doesn't have a great way to provide multiple-inheritance/mixin-based types.

However, it does seem that all `Type[r.Resource]` objects _also_ have the `DataclassDictMixIn` mixin - so to help the typing system figure things out, I added this `issubclass` check.

Reply:
I think we can just remove this test as it was not present before as well and anyway there is going to be a failure if `from_dict` is not defined (which should never be the case using standard resources or generic resources).